### PR TITLE
Replacing embedded-consul with testcontainers

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**
Motivation: #95

### CODE REVIEW

File: src/test/java/io/vertx/ext/consul/suite/Checks.java [hunk 1]
```diff
@@ -55,16 +55,16 @@ public void bindCheckToService() {
       .setServiceId(serviceId)
       .setStatus(CheckStatus.PASSING)
       .setTtl("10s");
-    runAsync(() -> ctx.writeClient().registerCheck(check));
+    runAsync(() -> masterClient.registerCheck(check));
```
Comment (ruslansennov) [inline src/test/java/io/vertx/ext/consul/suite/Checks.java:58]: why the client is changed?

File: src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    }
+
+    public Builder join(ConsulInstance other) {
+      Objects.requireNonNull(other);
+      this.join = true;
+      if (isMacOS()) {
+        this.serverAddr = other.getContainerNetwork().getIpAddress();
+        this.serverSerfLanPort = Builder.SERF_LAN_PORT;
+      } else {
+        //подключение по адерсу gateway и внешнему порту
```
Comment (ruslansennov) [inline src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java:175]: translate plz

File: src/test/resources/health_http_verticale.groovy
Comment (ruslansennov) [inline src/test/resources/health_http_verticale.groovy:1]: verticle

File: src/test/java/io/vertx/ext/consul/vertx/VertxHttpServer.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.MountableFile;
+
+import java.time.Duration;
+
+public class VertxHttpServer extends GenericContainer<VertxHttpServer> {
+  private static final String IMAGE = "vertx/vertx4";
+  private final String VERTICLE_HOME = "/usr/verticles";
+  private final String STATUS_FILE = "health_status.txt";
+  private final String HEALTH_HTTP_VERTICLE = "health_http_verticale.groovy";
```
Comment (ruslansennov) [inline src/test/java/io/vertx/ext/consul/vertx/VertxHttpServer.java:17]: dito

File: src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      .setHost(address())
+      .setPort(secure ? getMappedPort(Builder.HTTPS_PORT) : getMappedPort(Builder.HTTP_PORT))
+      .setSsl(secure);
+  }
+
+  public ConsulClient createSecureClient(
+    Vertx vertx, String token, boolean trustAll, PemTrustOptions trustOptions
+  ) {
+    ConsulClientOptions options = consulClientOptions(token, true)
+      .setHost("127.0.0.1")
```
Comment (ruslansennov) [inline src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java:93]: why does this work with '127.0.0.1'?

**Review by ruslansennov — CHANGES_REQUESTED** [review]

File: src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java (hunk 4 above)
Comment (ViktoriiaLebedeva) [reply src/test/java/io/vertx/ext/consul/instance/ConsulInstance.java]: It's a default address. I fixed it, but there was an error "No subject alternative names matching IP address ...". Setting VerifyHost to false helped.

**Pull request merged**